CONTRIBUTING.md

```diff
@@ -0,0 +1,105 @@
+
+
+ - [Code of Conduct](#coc)
+ - [Question or Problem?](#question)
+ - [Issues and Bugs](#issue)
+ - [Feature Requests](#feature)
+ - [Submission Guidelines](#submit)
+
+## <a name="coc"></a> Code of Conduct
+Help us keep the project open and inclusive. Please read and follow our (Code of Conduct)[https://github.com/UdacityMobileWebScholarship/showcase-app/blob/master/CODE_OF_CONDUCT.md].
+
+## <a name="question"></a> Got a Question, Query or Problem?
+
+Do not open issues for general queries as we want to keep GitHub issues for bug reports, feature requests and project development related contributions. For general queries reach out to us on [#cp_shwcase_app](https://mobilewebindiascholar.slack.com/messages/CA8MRBPAT) channel on slack.
+
+## <a name="issue"></a> Found a Bug?
+If you find a bug in the source code, you can help us by
+[submitting an issue](https://github.com/UdacityMobileWebScholarship/showcase-app/issues/new) to our GitHub Repository. Even better, you can check out issues [open for claim](https://github.com/UdacityMobileWebScholarship/showcase-app/labels/open%20for%20claim) and claim one!
+
+## <a name="feature"></a> Missing a Feature?
+You can *request* a new feature by [submitting an issue](https://github.com/UdacityMobileWebScholarship/showcase-app/issues/new) to our GitHub
+Repository. Given the project timeline and necessity for the feature, we'll discuss the roadmap accordingly. If you would like to *implement* a new feature, please submit an issue with
+a proposal for your work first, to be sure that we can use it.
+
+## <a name="submit"></a> Submission Guidelines
+
+### <a name="submit-issue"></a> Submitting an Issue
+Before you submit an issue, please search the issue tracker, maybe an issue for your identified problem already exists and you can join the discussion there only.
+For bugs, please try to provide as much information possible to reproduce it as possible, that will really help us a lot!
+
+### <a name="submit-pr"></a> Submitting a Pull Request (PR)
+1. Make sure there is an issue open for the PR and it is assigned to you before starting any work. That way it'll be helpful to eliminate duplicate efforts.
+2. Fork the repo.
+3. Make your changes in a new git branch:
+
+     ```shell
+     git checkout -b my-fix-branch master
+     ```
+4. Create your patch, **including appropriate test cases**.
+5. Run the full project test suite, as described in the [readme](https://github.com/UdacityMobileWebScholarship/showcase-app/tree/master#user-content-project-set-up),
+  and ensure that all tests pass.
+6. Commit your changes using a descriptive commit message. ex-
+```shell
+    git commit -m "doc: update readme"
+    git commit -m "ref: refactor API service"
+    git commit -m "feat: add authentication service"
```

Comment:
I am actually against commit messages in this tense. The messages should look like `Updates readme`, `Refactors API Service`, `Adds authentication service`. I hope you get my point.

Reply:
Actually, new collaborators tend to follow what is written in the Contribution Guidelines. We should also add links to some commit best practices.

Reply:
That would be great! Can you add some or suggest so I can add!

Reply:
Git branch naming practices can also be included to segregate Work in Progress(wip), bug, feature etc. [Source](https://stackoverflow.com/questions/273695/git-branch-naming-best-practices?utm_medium=organic&utm_source=google_rich_qa&utm_campaign=google_rich_qa)

Reply:
I mostly use `issuenumber-someverbosetext` as branch names. It helps me remember which issue I am addressing. Like `1852-check-input-type-select-atoms`.

Reply:
I like the idea of issue number prefix!

Reply:
@PrabhanshuAttri `wip:`, `feat:` can be included in commit msg as well, it makes more sense there.
adding labels to issues/PR can help us filter category wise!

Reply:
@ritikrishu Exactly. The commit message should ideally look like this.

```
A small 50 character line

Long description describing the change
with keywords for filtering and all
```

For that we need to enforce using `git commit` without `-m` switch. The collaborator can use `git commit -m` when the information is small and can be written in less than 50 characters.

Reply:
this is a minimal example for those who are not well well versed with git. We'd want them to learn along the way. Contributors can always look into other PRs who've been merged for reference. We can even add one or two links in there when we have such code contribution PRs.